SU2_CFD/include/fluid/CCoolProp.hpp

```diff
@@ -0,0 +1,139 @@
+/*!
+ * \file CCoolProp.hpp
+ * \brief Defines the state-of-the-art fluid model from CoolProp library.
+ * \author P. Yan, G. Gori, A. Guardone
+ * \version 7.3.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2022, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include "CFluidModel.hpp"
+#include "CoolProp.h"
+#include "AbstractState.h"
+#include "crossplatform_shared_ptr.h"
+
+
+/*!
+ * \class CCoolProp
+ * \brief Child class for defining fluid model from CoolProp library.
+ * \author: P.Yan
+ */
+
+class CCoolProp final : public CFluidModel {
+    protected:
+        su2double Gamma{0.0};           /*!< \brief Ratio of Specific Heats. */
+        su2double Gas_Constant{0.0};    /*!< \brief specific Gas Constant. */
+        su2double Pressure_Critical{0.0};   /*!< \brief critical pressure */
+        su2double Temperature_Critical{0.0};    /*!< \brief critical temperature */
+        shared_ptr<CoolProp::AbstractState> fluid_entity;   /*!< \brief fluid entity */
+
+public:
+/*!
+ * \brief Constructor of the class.
+ */
+CCoolProp(string fluidname);
+
+/*!
```

Comment:
change done

Reply:
In the future use github to apply the suggestions, there are buttons to "commit suggestion" or "add suggestion to batch" to commit multiple suggestions.
This will also resolve the conversation and makes it easier to track what was addressed